<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
	<title>BMI 214 Main/Home Page</title>
	<link rel="stylesheet" type="text/css" href="http://bmi214.stanford.edu/pub/skins/enlighten/default.css" media="screen" title="style (screen)" />
	<link rel="stylesheet" type="text/css" href="http://bmi214.stanford.edu/pub/skins/enlighten/brown/color.css" media="screen" title="style (screen)" />
<!--HTMLHeader--><style type='text/css'><!--
span.anchor {
	float: left;
	font-size: 10px;
	margin-left: -10px;
	width: 10px;
    position:relative; top:-0.1em;
	text-align: center;
}
span.anchor a { text-decoration: none; }
span.anchor a:hover { text-decoration: underline; }
ol.toc { text-indent:-20px; list-style: none; }
ol.toc ol.toc { text-indent:-40px; }
div.tocfloat { font-size: smaller; margin-bottom: 10px;
    border-top: 1px dotted #555555; border-bottom: 1px dotted #555555;
    padding-top: 5px; padding-bottom: 5px; 
    width: 38%; float: right; margin-left: 10px; clear: right;
    margin-right:-13px; padding-right: 13px; padding-left: 13px;
    background-color: #eeeeee; }
div.toc { font-size: smaller; 
    padding: 5px; border: 1px dotted #cccccc;
    background: #f7f7f7;
    margin-bottom: 10px; }
div.toc p { background-color: #f9f6d6;
    margin-top:-5px;   padding-top: 5px;
    margin-left:-5px;  padding-left: 5px;
    margin-right:-5px; padding-right: 5px;
    padding-bottom: 3px;
    border-bottom:  1px dotted #cccccc; }.editconflict { color:green; 
  font-style:italic; margin-top:1.33em; margin-bottom:1.33em; }
#logo-box .sitetitle a{ height:0px; background: url(http://helix-web.stanford.edu/bmi214/images/smi_icon.gif) left 16px no-repeat; } #logo-box .sitetitle a, #logo-box .sitetag{ padding-left: 0px; } #wrap { width: 850px; } #sidebar { width: 212px; margin-left: -222px; } #content { margin-right: 214px; } #header .headerphoto { background: #FFF url(http://bmi214.stanford.edu/pub/skins/enlighten/headerphoto.jpg) no-repeat; } 
  table.markup { border:2px dotted #ccf; width:90%; }
  td.markup1, td.markup2 { padding-left:10px; padding-right:10px; }
  table.vert td.markup1 { border-bottom:1px solid #ccf; }
  table.horiz td.markup1 { width:23em; border-right:1px solid #ccf; }
  table.markup caption { text-align:left; }
  div.faq p, div.faq pre { margin-left:2em; }
  div.faq p.question { margin:1em 0 0.75em 0; font-weight:bold; }
  div.faqtoc div.faq * { display:none; }
  div.faqtoc div.faq p.question 
    { display:block; font-weight:normal; margin:0.5em 0 0.5em 20px; line-height:normal; }
  div.faqtoc div.faq p.question * { display:inline; }
   
    .frame 
      { border:1px solid #cccccc; padding:4px; background-color:#f9f9f9; }
    .lfloat { float:left; margin-right:0.5em; }
    .rfloat { float:right; margin-left:0.5em; }
a.varlink { text-decoration:none; }

--></style>
  <link rel='stylesheet' href='http://bmi214.stanford.edu/pub/wsplus/wsplus.css' 
    type='text/css' />
  <!--[if IE]><style type='text/css' media='screen'>
    body { behavior:url('http://bmi214.stanford.edu/pub/wsplus/csshover.htc'); }
    .rollover * { visibility: visible; }
  </style><![endif]-->
<script type="text/javascript">
function toggle(obj) {
    var elstyle = document.getElementById(obj).style;
    var text    = document.getElementById(obj + "tog");
    if (elstyle.display == 'none') {
        elstyle.display = 'block';
        text.innerHTML = "hide";
    } else {
        elstyle.display = 'none';
        text.innerHTML = "show";
    }
}
</script>  <meta name='robots' content='index,follow' />

</head>
<body>
<div id="wrap">
	<div id="header">
		<div class="pageactions">
			<!--PageActionFmt-->
			<ul><li class='print'>     <a accesskey=''  rel='nofollow'  class='wikilink' href='http://bmi214.stanford.edu/index.php?n=Main.HomePage?action=print'>Print</a>
</li></ul>

			<!--/PageActionFmt-->
		</div>

		<div id="logo-box">
			<!--PageHeaderFmt-->
			<h1 class='sitetitle'><a class='urllink' href='http://bmi214.stanford.edu/index.php' title='' rel='nofollow'>BMI 214</a></h1>
<div class='sitetag' >Representations and Algorithms for Computational Molecular Biology
</div>

			<!--/PageHeaderFmt-->
		</div>
		<div class="headerphoto"></div>
	</div>

	<div id="menu">
		<!--PageTabsFmt-->
		<ul><li><a class='selflink' href='http://bmi214.stanford.edu/index.php?n=Main.HomePage'>Home</a>
</li><li><a class='wikilink' href='http://bmi214.stanford.edu/index.php?n=Site.Calendar'>Calendar</a>
</li><li><a class='wikilink' href='http://bmi214.stanford.edu/index.php?n=Site.Assignments'>Assignments</a>
</li><li><a class='wikilink' href='http://bmi214.stanford.edu/index.php?n=Site.CodePolicy'>Code Policy</a>
</li><li><a class='wikilink' href='http://bmi214.stanford.edu/index.php?n=Site.Grades'>Grades</a>
</li><li><a class='wikilink' href='http://bmi214.stanford.edu/index.php?n=Site.PythonTutorial'>Python Tutorial</a>
</li><li><a class='wikilink' href='http://bmi214.stanford.edu/index.php?n=Site.Readings'>Readings</a>
</li><li><a class='wikilink' href='http://bmi214.stanford.edu/index.php?n=Site.Videos'>Videos</a>
</li><li>
</li></ul>

		<!--/PageTabsFmt-->
	</div>

	<div id="body-wrap">
		<div id="content-wrap"><div id="content">
			<div id="main">
				<!--PageTitleGroupFmt-->
				<a href='http://bmi214.stanford.edu/index.php?n=Main' class="pagegroup">Main</a>
				<!--/PageTitleGroupFmt-->
				<!--PageTitleFmt-->
				<h2 class='page-head'><a href='http://bmi214.stanford.edu/index.php?n=Main.HomePage'>Home Page</a></h2>
				<!--/PageTitleFmt-->

				<!--PageText-->
<div id='wikitext'>
<h4>Biomedical Informatics 214:<br />Representations and Algorithms for Computational Molecular Biology<br />(also listed as Bioengineering 214, Computer Science 274 and Genetics 214)</h4>
<p><a name='geninfo' id='geninfo'></a>
<a href='#description'>Description</a> | <a href='#staff'>Staff</a> | <a href='#piazza'>Announcements / Discussion</a> | <a href='#units'>Units</a> | <a href='#grading'>Grading</a> | <a href='#codepolicy'>Code Policy</a> | <a href='#exams'>Exams</a> | <a href='#latepolicy'>Late Policy</a> | <a href='#partnerpolicy'>Partner Policy</a> | <a href='#honor'>Honor Code</a> | <a href='#auditor'>Auditors</a> | <a href='#prereq'>Prerequisites</a> | <a href='#compres'>Computer Resources</a> | <a href='#accessibility'>Accessibility</a> | <a href='#textbook'>Textbook</a> | <a href='#note'>Note on Comp Bio Courses</a>
</p>
<p class='vspace'><a name='description' id='description'></a><strong>Description</strong>: (<a href='#geninfo'>top</a>)
</p>
<div class='vspace'></div><div class='indent'>This course will introduce the basic computational methods used in molecular biology, combining core lectures, programming assignments, with midterm and final. The course will introduce and use biological data sources available on the internet. Topics will include basic algorithms for analysis of sequence, structure and function, particularly including alignment of biological sequences and structures, as well as more advanced representational and algorithmic issues. These include, for example, dynamic programming algorithms for alignment, structural superposition algorithms, computing with distance information, 1D and 3D motif definition and computation, hidden Markov models, phylogenetic trees, statistical feature detection, RNA sequence and structure, chemoinformatics and network analysis.
</div><p class='vspace'><a name='staff' id='staff'></a><strong>Instructor:</strong>
</p>
<div class='vspace'></div><div class='indent'><a class='urllink' href='http://bmi214.stanford.edu/people/altman/index.html' title='' rel='nofollow'>Russ B. Altman</a> <br />Professor of Bioengineering, Genetics, &amp; Medicine (and Computer Science by courtesy)  <br />russ.altman at stanford.edu
</div><p class='vspace'><strong>Location:</strong>
</p><div class='indent'><strong>Lectures</strong>: Tuesdays and Thursdays, 4:15pm-5:30pm, Gates B3<br /><br /><strong>Sections</strong>: A few Fridays, 3:15pm-4:05pm, Gates B3 <br />Week 1 (September 26): Python Tutorial 1 (of 2) - Basic syntax and usage <br />Week 2 (October 3): Python Tutorial 2 (of 2) - Slightly more advanced topics <br />
</div><p><strong>Course Coordinator:</strong>
</p>
<div class='vspace'></div><div class='indent'>Tiffany Murray (tiffany.murray at stanford.edu)  <br />Department of Bioengineering <br />Shriram Center, 443 Via Ortega Room 213 <br />650-725-0659
</div><p class='vspace'><strong>Teaching Assistants:</strong>
</p>
<div class='vspace'></div><div class='indent'>Kun-Hsing Yu <br />Office Hours: Mondays, 2:30-3:30 PM, Medical School Office Building X237 <br /><strong>Office hour before midterm: Thursday Oct. 30 4:15-5:30 PM, Gates B3</strong> <br /><strong>No office hour on Nov. 3.</strong> <br /><br />Tomer Altman <br />Office Hours: (starting October 28th) <br />Tuesdays, 1:30-2:30 PM, Medical School Office Building, X237 <br />Simulcasted using Google Hangouts <br /><br />Winn Haynes <br />Office Hours: Wednesdays, 1-2 PM, Location: <br />- Oct 1 and Oct 8: Medical School Office Building X138 <br />- Oct 22 and after: Medical School Office Building X237 <br /><br />Emily Tsang <br />Office Hours: Fridays, 3-4 PM, Medical School Office Building X237<br /><strong>Friday Dec. 5: 1:30-2:30 PM, MSOB X237</strong><br /><br />
</div><div  style='display: none;' > 
<div class='indent'>David Poznik <br />Office Hours: Mondays, 4:15-5:15pm, Littlefield 301 (southeast corner) <br /><strong>Note:</strong> Littlefield is located near the oval. If 301 is occupied, there will be a note on the door directing you to another room on the third floor.<br /><br />Collin Melton <br />Office Hours: Tuesdays, 9:50am-10:50am, MSOB X271 <br /><br />Kun-Hsing Yu <br />Office Hours: Wednesdays, 10:30am-11:30am, Huang 16 <br /><br />Sameer Arya <br />Office Hours: Thursdays, 2:15pm-3:15pm, MSOB X275 <br /><br />Linda Szabo <br />Office Hours: Fridays, 9am-10am, MSOB X275
</div></div>
<div class='vspace'></div><div class='indent'><strong>Contacting the TAs</strong>: The staff mailing list is biomedin214-aut1415-staff at lists.stanford.edu. This list is for <strong>personal matters only</strong>. The staff will not respond to non-private questions on the mailing list. Please post them to Piazza instead (see below).
</div><p class='vspace'><a name='piazza' id='piazza'></a><strong>Announcements / Discussion:</strong>(<a href='#geninfo'>top</a>)
</p>
<div class='vspace'></div><div class='indent'><strong>Announcements</strong> <br />All announcements will be posted as Instructor Notes on the Piazza forum: <br /><a class='urllink' href='https://piazza.com/stanford/fall2014/biomedin214/' title='' rel='nofollow'>https://piazza.com/stanford/fall2014/biomedin214/</a> <br />Sign up now. <br /><br /><strong>Discussion</strong> <br />Please post questions about assignments to the Piazza forum so that all students can benefit from the answers. Students are encouraged to answer each others' questions, and participation points will be awarded accordingly. The TAs will peruse these pages to answer clarification questions and to endorse student answers. <br />
</div><p class='vspace'><a name='units' id='units'></a><strong>Units:</strong>(<a href='#geninfo'>top</a>)
</p>
<p class='vspace'>Biomedin 214
</p><ul><li>This course is normally taken for 4 units.
</li><li>It can be taken for 3 units by arrangement with instructor ONLY.
</li></ul><p>Biomedin 216 [by arrangement with instructor only]<br />Students must attend all lectures; absences must be approved by the instructor. 
</p><ul><li>1 unit: lectures only
</li><li>2 units: lectures, assignments, midterm, final
</li></ul><p class='vspace'><a name='grading' id='grading'></a><strong>Grading</strong>: (<a href='#geninfo'>top</a>)
</p>
<div class='vspace'></div><div class='indent'>The course will be graded by performance on 3 short homework assignments (15%), 4 programming projects (60%), a midterm (10%), a final (10%), and participation (5%).
</div><p class='vspace'>Participation is based on:
</p><ul><li>Attending class. To receive credit for attendance, you must physically sign the sheet. Please do not email later to request a retroactive sign-in.
</li><li>Asking questions in class. Please identify yourself to Dr. Altman.
</li><li>Contributing to the Piazza newsgroup by asking and/or answering good questions.
</li></ul><p class='vspace'><a name='codepolicy' id='codepolicy'></a><strong>Code / Language Policy</strong>: (<a href='#geninfo'>top</a>)
</p>
<div class='vspace'></div><div class='indent'>Familiarize yourself with the <a class='wikilink' href='http://bmi214.stanford.edu/index.php?n=Site.CodePolicy'>Code/Language Policy</a> before starting the first programming assignment.
</div><p class='vspace'><a name='exams' id='exams'></a><strong>Exams</strong>: (<a href='#geninfo'>top</a>)
</p>
<div class='vspace'></div><div class='indent'>Both exams are open-notes. No internet.
</div><div class='indent'><strong>Midterm</strong>: 6:00-7:30 pm, Monday, November 3 (Week 7) LKSC 120 and 130.
</div><div class='indent'><strong>Final</strong>: 12:15-3:15 pm, Monday, December 8 @ LK101/102 
</div><div  style='display: none;' > 
<dl><dd><div class='indent'>Last name A - F: Alway M208
</div><div class='indent'>Last name G - Z: Alway M114
</div></dd></dl></div>
<p class='vspace'><a name='latepolicy' id='latepolicy'></a><strong>Late Policy</strong>: (<a href='#geninfo'>top</a>)
</p>
<div class='vspace'></div><div class='indent'>Each student is granted 168 "free" late hours that can be used as extensions for any project or assignment. This is <strong>a total of 168 hours</strong> for the entire quarter, not per assignment. Late time will be measured with no distinction for weekends or holidays, and will be rounded UP to the nearest integer (thus, 10 minutes late = 1 hour late). After you use up all your free hours, your grade on late projects/assignments will be reduced by 0.5 percentage points for each late hour. So if your project is graded as 85 (of 100) but was turned in 24 hours late (beyond the free 168), we record a 73. <br /><strong>The clock runs the same for everyone, even those who join the course late.</strong>
</div><p class='vspace'><a name='partnerpolicy' id='partnerpolicy'></a><strong>Partner Policy</strong>: (<a href='#geninfo'>top</a>)
</p>
<div class='vspace'></div><div class='indent'>For assignments:  <br />Students may discuss and work on problems in groups but must write up their own solutions. When writing up the solutions, students must write the names of people with whom they discussed the assignment. <br /><br />For programming projects:  <br />Students may discuss ideas with others. However, programs are to be completed independently and should be original work. Posting or sharing code, such as on Piazza or in a public repository, will be considered a violation of the Honor Code. <strong>If students choose to use a version control system such as GitHub, it is their responsibility to make sure that permission is set so that their code is not accessible by anyone else.</strong> Names of students with whom programming ideas were discussed should be included with assignments and explicitly indicated in the header comments of all source code files.<br /><br /><strong>If you do not list a particular student, this will be interpreted as an attestation that you did not speak with this student about the assignment or project.</strong>
</div><p class='vspace'><a name='honor' id='honor'></a><strong>Honor Code</strong>: (<a href='#geninfo'>top</a>)
</p><div class='indent'>Students must abide by the terms of the <a class='urllink' href='http://studentaffairs.stanford.edu/communitystandards/policy/honor-code' title='' rel='nofollow'>Stanford Honor Code</a>.
</div><p class='vspace'><a name='auditor' id='auditor'></a><strong>Auditors</strong>: (<a href='#geninfo'>top</a>)
</p>
<div class='vspace'></div><div class='indent'>Auditors for the course should take it for one unit as BMI 216. This course requires attendance (and sign-in) at each lecture, but does not require completion of homeworks or exams. Auditors who want to sit-in on the course but not be officially signed up for 1 unit of credit should get approval from Dr. Altman, and will also be asked to attend all lectures and sign-in.
</div><p class='vspace'><a name='prereq' id='prereq'></a><strong>Prerequisites</strong>: (<a href='#geninfo'>top</a>)
</p>
<div class='vspace'></div><ol><li>Programming skills are required at the level of <a class='urllink' href='http://www.stanford.edu/class/cs106b/' title='' rel='nofollow'>CS106B</a> or <a class='urllink' href='http://www.stanford.edu/class/cs106x/' title='' rel='nofollow'>CS106X</a>. This course has a significant programming component, so students should enter it with the ability to create moderately complex data structures and implement algorithms using them. <a class='urllink' href='http://www.stanford.edu/class/cs161/' title='' rel='nofollow'>CS161</a> and <a class='urllink' href='http://www.stanford.edu/class/cs108/' title='' rel='nofollow'>CS108</a> would also be great, but you should be OK without them. Students who have attempted to take BMI214 with just CS106A under their belts have struggled, so <em>caveat emptor</em>. In particular, we highly recommend a good understanding of recursion for the first programming project, and the better a feel you have for classes, functions, and standard data structures (lists, dictionaries/maps, sets, etc), the easier a time you'll have. Acceptable languages are outlined in the code policy. If you're comfortable with the material of CS106B, but have never used Python, you'll be fine. However, we strongly advise that you learn it as soon as possible so that when you're working on the projects, you can focus on the algorithms rather than the syntax. <strong>Treat learning Python as your homework for the first week of class!</strong> Project 1 can be tricky to debug, and the more comfortable you are with Python, the fewer bugs you'll have to find :)
<div class='vspace'></div></li><li><a class='urllink' href='http://www.stanford.edu/dept/biology/indexacademics.html' title='' rel='nofollow'>Biology 40</a> or equivalent is recommended, since we will quickly move through many biology topics. We recommend that all students page through the Biology Tutorial on the sidebar of this site. For many students, this will be a quick review, but it might take an hour or two for those who do not think about biology very much.
</li></ol><p class='vspace'><a name='compres' id='compres'></a><strong>Computer Resources</strong>: (<a href='#geninfo'>top</a>)
</p>
<div class='vspace'></div><ul><li>You will need to use your SUNet ID. If you don't have a SUNet ID, see <a class='urllink' href='http://sunetid.stanford.edu' title='' rel='nofollow'>http://sunetid.stanford.edu</a>
</li><li>Make sure that you are registered for the course on Axess so that you will receive email announcements sent to the course list
</li><li>You will need to have access to email, the course website, and the Stanford FarmShare computing resource. All of these resources are available at Stanford computer labs (such as at Sweet Hall) as well as through remote access (SSH or VPN). 
</li></ul><div class='vspace'></div><div class='indent'>The Stanford FarmShare <code>corn</code> cluster is available for running compute-intensive jobs for the course. To log in to the <code>corn</code> cluster machines, use a secure shell (ssh) client.
<div class='vspace'></div></div><div class='indent'>On Windows: You will have to download a terminal emulation that allows ssh. Stanford offers a few free ones <a class='urllink' href='http://www.stanford.edu/services/ess/pc/index.html' title='' rel='nofollow'>here</a>; a popular one is <a class='urllink' href='http://www.chiark.greenend.org.uk/~sgtatham/putty/download.html' title='' rel='nofollow'>PuTTY</a>. Directions for using Putty to connect to <code>corn</code>:
<div class='indent'>1. Under "Host Name", enter <code>corn.stanford.edu</code>
</div><div class='indent'>2. Under "Protocol", choose SSH
</div><div class='indent'>3. Press the "open" button.
</div><div class='indent'>A terminal window should appear, connected to <code>corn</code>. Putty will tell you if there was an error.
</div><div class='vspace'></div></div><div class='indent'>On Mac OS X, Unix, Linux:
<div class='indent'>1.  Open a terminal window
</div><div class='indent'>2. Type <code>ssh -X sunetid@corn.stanford.edu</code>
</div><div class='vspace'></div></div><div class='indent'>NOTE: If you are unfamiliar with the Unix command-line, this is a helpful list of basic commands:
</div><div class='indent'><a class='urllink' href='http://bmi214.stanford.edu/files/fwunixref.pdf' title='' rel='nofollow'>Unix Reference</a><br />
<div class='vspace'></div></div><div class='indent'>For more information on the various campus computers you can access:
<div class='indent'><a class='urllink' href='http://www.stanford.edu/services/cluster/' title='' rel='nofollow'>http://www.stanford.edu/services/cluster/</a> 
</div><div class='indent'><a class='urllink' href='http://www.stanford.edu/services/cluster/which.html' title='' rel='nofollow'>http://www.stanford.edu/services/cluster/which.html</a>
</div><div class='vspace'></div></div><div class='indent'>You will also need to be able to transfer files to the corn machines using SCP.  If you are on a computer with command line <code>scp</code> you can transfer a file, assignment.zip with:
<div class='vspace'></div><div class='indent'><code class='escaped'> $ scp assignment.zip sunetid@corn.stanford.edu:~/ </code><br />
</div><div class='vspace'></div></div><div class='indent'>Or there are GUI sftp clients files available:
<div class='indent'>For Mac OS X: <a class='urllink' href='http://cyberduck.ch/' title='' rel='nofollow'>http://cyberduck.ch/</a>
</div><div class='indent'>For Windows: <a class='urllink' href='http://winscp.net/eng/index.php' title='' rel='nofollow'>http://winscp.net/eng/index.php</a>
</div><div class='vspace'></div></div><div class='indent'>There are also other options available from the stanford IT web sites:
<div class='indent'><a class='urllink' href='http://www.stanford.edu/group/security/securecomputing/sftp.html' title='' rel='nofollow'>http://www.stanford.edu/group/security/securecomputing/sftp.html</a>
</div><div class='indent'><a class='urllink' href='https://itservices.stanford.edu/service/ess' title='' rel='nofollow'>https://itservices.stanford.edu/service/ess</a>
</div><div class='vspace'></div></div><div class='indent'>Some course material will be placed on the course website in *.pdf (Adobe Acrobat) format, which allows the documents to be read on multiple platforms. Readers are available for free for Windows, Macintosh and many Unix platforms at the <a class='urllink' href='http://www.adobe.com/products/acrobat/' title='' rel='nofollow'>Adobe website</a>.<br />
</div><p class='vspace'><a name='accessibility' id='accessibility'></a><strong>Students with Documented Disabilities</strong>: (<a href='#geninfo'>top</a>)
</p>
<div class='vspace'></div><div class='indent'>Students who may need an academic accommodation based on the impact of a disability must initiate the request with the Office of Accessible Education (OAE). Professional staff will evaluate the request with required documentation, recommend reasonable accommodations, and prepare an Accommodation Letter for faculty dated in the current quarter in which the request is being made. Students should contact the OAE as soon as possible since timely notice is needed to coordinate accommodations. The OAE is located at 563 Salvatierra Walk; phone: 723-1066; URL: <a class='urllink' href='http://studentaffairs.stanford.edu/oae' title='' rel='nofollow'>http://studentaffairs.stanford.edu/oae</a>.<br /><br />
</div><p><a name='textbook' id='textbook'></a><strong>Optional Course Textbook</strong>: (<a href='#geninfo'>top</a>)
</p>
<div class='vspace'></div><ul><li>Durbin, R., Eddy, S.R., Krogh, A., Mitchison, G., <strong><em><a class='urllink' href='http://www.cambridge.org/asia/catalogue/catalogue.asp?isbn=9780521629713' title='' rel='nofollow'>Biological Sequence Analysis : Probabilistic Models of Proteins and Nucleic Acids</a></em></strong>. 1999, Cambridge Univ Pr. ISBN: 0521629713'''
</li></ul><p class='vspace'>Other Recommended books:
</p>
<div class='vspace'></div><ul><li>Beazley, David M., <strong><em><a class='urllink' href='http://www.amazon.com/Python-Essential-Reference-Developers-Library/dp/0672328623' title='' rel='nofollow'>Python Essential Reference, 3rd ed.</a></em></strong>, SAMS Publishers, 2006. Chapter 1 is an excellent tutorial and introduction to Python, and overall, this can be a valuable reference when coding.
<div class='vspace'></div></li><li>Kohane, I.S., Kho, A., Butte, A.J., <strong><em><a class='urllink' href='http://mitpress.mit.edu/catalog/item/default.asp?sid=644AC585-7E48-4842-B043-8682692F15C8&amp;ttype=2&amp;tid=9003' title='' rel='nofollow'>Microarrays for an Integrative Genomics (Computational Molecular Biology)</a></em></strong>. 2002, MIT Press. ISBN: 026211271X. 
<div class='vspace'></div></li><li>Mount, D.W., <strong><em><a class='urllink' href='http://www.bioinformaticsonline.org/' title='' rel='nofollow'>Bioinformatics : sequence and genome analysis</a></em></strong>. 2nd edition (July, 2004), Cold Spring Harbor Laboratory Press. ISBN: 0879696877.
<div class='vspace'></div></li><li>Bourne, P.E., Weissig, H. (editors), <strong><em><a class='urllink' href='http://www.wiley.com/cda/product/0,,0471202002,00.html' title='' rel='nofollow'>Structural Bioinformatics</a></em></strong>. 2004, John Wiley &amp; Sons. ISBN: 0471201995. This book is also available from the Wiley Interscience website at <a class='urllink' href='http://www3.interscience.wiley.com/cgi-bin/homepage/?isbn=0471721204' title='' rel='nofollow'>http://www3.interscience.wiley.com/cgi-bin/homepage/?isbn=0471721204</a> via the campus network.
</li></ul><div class='vspace'></div><hr />
<p class='vspace'><a name='note' id='note'></a><strong>Note on courses in computational biology</strong>: (<a href='#geninfo'>top</a>)
</p>
<div class='vspace'></div><ul><li>BMI 214 (also listed as CS 274, BioE 214, Genetics 214) is this course. It has been taught since 1996 and is an introduction to representations and algorithms for analysis of sequence, structure and function. It requires programming skills and aims to give an understanding of the molecular biological problems that arise, and how algorithms are developed to address them. It does not train students to be expert users of tools, but gives them an in-depth knowledge of some tools and a broad introduction to the technical issues in analysis of biological data.
</li></ul><p class='vspace'>There are several others courses at Stanford that may cover some overlapping material, but every instructor explains things differently and with different emphasis (all opinions are mine, and they may be wrong--I haven't taken any of these classes, just read about them and talked a bit to instructors, so talk to someone who has!)
</p>
<div class='vspace'></div><ul><li>BMI 217 (Atul Butte, Translational Bioinformatics):  covers some similar topics in RNA expression analysis (clustering/classification) and genome analysis, but links these more to clinical concepts, and also has an independent project at end.  Students routinely take this class and BMI 214.
<div class='vspace'></div></li><li>BMI 215 (Nigam Shah, Data Driven Medicine):  covers some Natural Language Processing, but not at the molecular level, also covers ontologies but this is not a focus of our course.  Students routinely take this class and BMI 214.
<div class='vspace'></div></li><li>BMI 258 (Biochemistry 158, Doug Brutlag, Genomics, Bioinformatics, and Medicine) is an introduction to genomics, bioinformatics and applications to medicine, and goes into more detail on the clinical implications of genomics.  <a class='urllink' href='http://bmi258.stanford.edu/' title='' rel='nofollow'>http://bmi258.stanford.edu/</a>
<div class='vspace'></div></li><li>BMI 231 (Biochemistry 218, Doug Brutlag, Computational Molecular Biology): (no longer offered, but course videos and lecture slides are available) this is more for tool users, but does cover some similar issues in genomics, sequence and structure analysis, in a complementary way to this class.  No programming required. Students who have taken this and BMI 214 report some overlap, but very different perspectives provided.  <a class='urllink' href='http://bmi231.stanford.edu/' title='' rel='nofollow'>http://bmi231.stanford.edu/</a>
<div class='vspace'></div></li><li>BMI 262 (CS 262, Serafim Batzoglou, Computational Genomics):  covers very similar concepts more deeply in sequence analysis from a computer science perspective (dynamic programming alignment, HMM, Gibbs sampling, context-free grammars, phylogenetics), but does not overlap very much in 3D structure or function analysis.  Students who have taken this and BMI 214 say that this is more rigorous in terms of algorithms, complexity, etc...and so somewhat complementary.
<div class='vspace'></div></li><li>BMI 273A (CS 173, CS 273A, Serafim Batzoglou &amp; Gill Bejerano, Computational Tour of Human Genome): both cover details of genome sequencing, genome organization and informatics analysis.  Programming encouraged but not required.  I do not have data about overlap of this class and BMI 214.
<div class='vspace'></div></li><li>Genetics 211 (Mike Cherry &amp; Gavin Sherlock):  This is an intro to genomics through Python programming and many genetics students have told me that it is a great warmup for BMI 214 if you are not as strong at programming.
</li></ul>
</div>

			</div>
		</div></div>

		<div id="sidebar" >
			<!--PageSearchFmt-->
			<p class="sidehead clear">Search</p>
			<form id="searchform" class="searchform" action='http://bmi214.stanford.edu/index.php' method="get"><fieldset><p>
				<input type='hidden' name='n' value='Main.HomePage' />
				<input type='hidden' name='action' value='search' />
				<input id='s' class='textbox' type='text' name='q' value="" />
				<input id='searchsubmit' class='button' type='submit' value='Go' />
			</p></fieldset></form>
			<!--/PageSearchFmt-->

			<!--PageRightFmt-->
			<ul><li><a class='wikilink' href='http://bmi214.stanford.edu/index.php?n=Site.BiologyTutorial'>Biology Tutorial</a>
</li><li><a class='wikilink' href='http://bmi214.stanford.edu/index.php?n=Site.SampleMidterm'>Sample Midterm Questions</a>
</li><li><a class='wikilink' href='http://bmi214.stanford.edu/index.php?n=Site.Jobs'>Jobs</a>
</li></ul><p class='vspace'  style='text-align: right;'> <span style='font-size:83%'><a class='wikilink' href='http://bmi214.stanford.edu/index.php?n=Site.SideBar?action=edit'>edit SideBar</a></span>
</p>
<div class='vspace'></div>

			<!--/PageRightFmt-->
		</div>
	</div>
	<div id="footer">
		<!--PageFooterFmt-->
			<p>Page last modified on November 28, 2014, at 06:53 PM
</p>

		<!--/PageFooterFmt-->
	</div>

</div>
</body>
</html>
